### PR TITLE
feat(config): extract gwas_significance parameter to step configuration

### PR DESCRIPTION
**Pull request by project-defiant**
## ✨ Context

https://github.com/opentargets/issues/issues/3327

## 🛠 What does this PR implement

This is the first part of the #3327. 
* Extraction of the `gwas_significance` (p-value) threshold parameter for the window-based-clumping step.
* Addition of `gwas_significance` to the airflow config with the new value (`1e-8`) overwriting default one (`5e-8`)
* Unified naming convention for the `WindowBasedClumpingStep` config to `WindowBasedClumpingStepConfig`

The main reason for this change is that we want to use the value of `1e-8` and possibly even more stringent value of `1e-9` in the future. 
## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: src/gentropy/config.py [hunk 1]
```diff
@@ -355,11 +355,17 @@ class VariantToGeneConfig(StepConfig):
...
     """Window-based clumping step configuration."""
 
+    session: Any = field(
+        default_factory=lambda: {
+            "start_hail": True,
+        }
+    )
     summary_statistics_input_path: str = MISSING
     study_locus_output_path: str = MISSING
+    gwas_significance: float = 5e-8
```
Comment (addramir) [inline src/gentropy/config.py:368]: Shouldn't it be 1e-8?

File: config/step/ot_window_based_clumping.yaml [hunk 2]
```diff
@@ -4,3 +4,4 @@ defaults:
 summary_statistics_input_path: ???
 study_locus_output_path: ???
 inclusion_list_path: ???
+gwas_significance: 1e-8
```
Comment (addramir) [inline config/step/ot_window_based_clumping.yaml:7]: What is the difference between configs here and WindowBasedClumpingStepConfig?

File: src/gentropy/method/window_based_clumping.py [hunk 3]
```diff
@@ -166,7 +168,16 @@ def clump(
 
         Returns:
             StudyLocus: clumped summary statistics (without locus collection)
+
+            Check WindowBasedClumpingStepConfig object for default values
         """
+        # p-value default value will change in the next major release
+        if gwas_significance == 5e-8:
+            warnings.warn(
+                "Default value for gwas_significance will change in 2.0 release",
```
Comment (addramir) [inline src/gentropy/method/window_based_clumping.py:177]: "Default value for gwas_significance will change in next release"

File: src/gentropy/window_based_clumping.py [hunk 4]
```diff
@@ -26,9 +29,12 @@ def __init__(
             summary_statistics_input_path (str): Path to the harmonized summary statistics dataset.
             study_locus_output_path (str): Output path for the resulting study locus dataset.
             distance (int): Distance, within which tagging variants are collected around the semi-index. Optional.
+            gwas_significance (float): GWAS significance threshold. Defaults to 5e-8.
```
Comment (addramir) [inline src/gentropy/window_based_clumping.py:32]: 1e-8?

**Comment by project-defiant** [comment]
@addramir just to clear things up here,
I wanted to persist the original value of 5e-8 in the gentropy API, as it can be used by downstream software and changing it might be a breaking change - so it should be introduced in a major version, thus 1.3 -> 2.0. It would also be good to inform our users about it before, so they can adjust to it.

The airflow layer that will run the data release will use the value of `gwas_significance` from [config/step/ot_window_based_clumping.yaml](https://github.com/opentargets/gentropy/pull/628/files/323d0c3bba17c3352a1edf9870fe75af8e86c9b4#diff-1ebc20fe481a0287b4a8f2dcb14b3e3058ee6dda23c790bbd962397f06e7a8fc)[config/step/ot_window_based_clumping.yaml](https://github.com/opentargets/gentropy/pull/628/files/323d0c3bba17c3352a1edf9870fe75af8e86c9b4#diff-1ebc20fe481a0287b4a8f2dcb14b3e3058ee6dda23c790bbd962397f06e7a8fc)
so we will still use value 1e-8 in the data release.

If you think we want to drop the original value, without maintaining the backwards compatibility for gentropy users,  I will drop the warning and change the default value in `WindowBasedClumpingStepConfig` to 1e-8.

**Review by ireneisdoomed — CHANGES_REQUESTED** [review]
Nice clean up! All good, 2 comments:
- I'd consider removing the warning.
- We apply a significance threshold also in the GWAS Catalog top hits ingestion. You might want to apply the same variable as well? Check [here](https://github.com/opentargets/gentropy/blob/fa6d500759d90cabdebac5124dd28da5d7f19be7/src/gentropy/datasource/gwas_catalog/associations.py#L1038C9-L1038C25).

**Comment by project-defiant** [comment]
@ireneisdoomed and @addramir thank you both for the feedback!

With regards to the issues, I would like to make a clear statments:
1. @addramir do we want to make threshold of 1e-8 as a default to all users that will use gentropy (even outside the Open Targets) - then it might be good to preserve the warning, otherwise I will drop it.
2. @addramir as @ireneisdoomed suggested, there is another threshold for the p-value found in [here](https://github.com/opentargets/gentropy/blob/fa6d500759d90cabdebac5124dd28da5d7f19be7/src/gentropy/datasource/gwas_catalog/associations.py#L1038C9-L1038C25) and is a part of the `GWASCatalogStudyInclusionGenerator`. The step is a part of the GWAS preprocessing DAG, which does not allow for the p-value change. Should we keep it in sync with the clumping step?
3. Looking over the code for 5e-8, there are a few more places - where the threshold is found 
* SummaryStatisticsQC
* SusieFineMapperStep
should we update these as well?

**Comment by addramir** [comment]
@project-defiant @ireneisdoomed Thank you for comments!
I think we need to go with 1e-8 everywhere. Since it is suggested as more correct default threshold. See for example paper from Peter Vischer:
https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1216-0#:~:text=We%20therefore%20recommend%20to%20use,increase%20of%20sample%20size%20%5B14%5D

"We therefore recommend to use a threshold of 1e-8 for GWAS with common variants, which might be slightly conservative for current datasets but should be appropriate for data from WGS or imputation-based studies in the future because the number of variants is expected to increase with the increase of sample size [[14](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1216-0#ref-CR14)]"

It is not really something to fight for really. The idea behind this data realise was to change the threshold only for GWAS Catalog for now. We can change it everywhere else (that is a correct thing) but we can do it later.

File: src/gentropy/method/window_based_clumping.py (hunk 3 above)
Comment (project-defiant) [reply src/gentropy/method/window_based_clumping.py]: I have dropped it

**Review by ireneisdoomed — APPROVED** [review]
👌

**Pull request merged**